parser/src/fstring.rs

```diff
@@ -209,6 +245,42 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_parse_fstring_nested_spec() {
+        let source = String::from("{foo:{spec}}");
+        let parse_ast = parse_fstring(&source).unwrap();
+
+        assert_eq!(
+            parse_ast,
+            FormattedValue {
+                value: Box::new(mk_ident("foo", 1, 1)),
+                conversion: None,
+                spec: Some(Box::new(FormattedValue {
+                    value: Box::new(mk_ident("spec", 1, 1)),
+                    conversion: None,
+                    spec: None,
+                })),
+            }
+        );
+    }
+
+    #[test]
+    fn test_parse_fstring_not_nested_spec() {
+        let source = String::from("{foo:spec}");
+        let parse_ast = parse_fstring(&source).unwrap();
+
+        assert_eq!(
+            parse_ast,
+            FormattedValue {
+                value: Box::new(mk_ident("foo", 1, 1)),
+                conversion: None,
+                spec: Some(Box::new(Constant {
+                    value: "spec".to_string(),
+                })),
```

Comment:
Could you add some more tests for some weirder cases? e.g. `f"{val: {'#'} }"` interprets the spec as `' {\'#\'} '`; it only interpolates the value if the only characters for the spec are `:{expression}}`

Reply:
I know this is not perfect yet but we can improve on that later. I added a few more fail cases tests.